File: Week03/LeetCode-1710.py
from functools import cmp_to_key
def compare(a, b):
    return b[1]-a[1]    # Decreasing order

class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        # Sort the boxes in decreasing oder with respecto to their units
        boxTypes.sort(key=cmp_to_key(compare))
        
        total = 0
        for noBoxes, noUnits in boxTypes:
            taken = min(noBoxes, truckSize)
            total += taken*noUnits
            truckSize -= taken
            if truckSize == 0: break
        return total
            

File: Week03/LeetCode-1710.cpp
using namespace:std;

class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        // Sort the boxes in decreasing order with respect to to their units
        sort(boxTypes.begin(), boxTypes.end(), 
                  [](vector<int> &a, vector<int> &b){
                            return a[1] > b[1];});
        int total = 0;
        for (int i=0; i<boxTypes.size(); i++){
            int noBoxes = boxTypes[i][0];
            int noUnits = boxTypes[i][1];            
            int taken = min(noBoxes, truckSize);
            total += taken*noUnits;
            truckSize -= taken;
            if (truckSize == 0) break;
        } // end-for
        
        return total;        
    } // end-maximumUnits
};